README.md
![](http://geo.azmag.gov/maps/readonaz/app/resources/img/maglogo_black.png)

# MAG Mapping Home Page #

[![Website](https://img.shields.io/website-up-down-green-red/http/shields.io.svg?label=my-website)](http://maps.azmag.gov/)
[![GitHub version](https://badge.fury.io/gh/AZMAG%2Fmap-MAGmappingPage.svg)](https://badge.fury.io/gh/AZMAG%2Fmap-MAGmappingPage)
[![GitHub issues](https://img.shields.io/github/issues/AZMAG/map-MAGmappingPage.svg)](https://github.com/AZMAG/map-MAGmappingPage/issues)
[![dependencies](https://david-dm.org/AZMAG/map-MAGmappingPage.png)](https://david-dm.org/AZMAG/map-MAGmappingPage)
[![devDependency Status](https://david-dm.org/AZMAG/map-MAGmappingPage/dev-status.png)](https://david-dm.org/AZMAG/map-MAGmappingPage)
[![Semver](http://img.shields.io/SemVer/2.0.0.png)](http://semver.org/spec/v4.4.7.html)
[![Built with Grunt](http://cdn.gruntjs.com/builtwith.png)](http://gruntjs.com/)
[![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](https://opensource.org/licenses/MIT)

The Maricopa Association of Governments (MAG) mapping home page is the portal to MAG's interactive mapping and analysis sites.

This JavaScript web-page was developed by MAG staff for MAG member agencies and the public.

## Keywords

`Maricopa County` `Pinal County` `State` `Arizona` `MAG` `maps` `ESRI` `#MAGmaps`

## Version

Releases will be numbered with the following format:

**`<major>.<minor>.<patch>`**

And constructed with the following guidelines:

1. **MAJOR** version when you make incompatible API changes **bumps the major** resets minor and patch
2. **MINOR** version when you add functionality in a backwards-compatible manner **bumps the minor** resets patch
3. **PATCH** version when you make backwards-compatible bug fixes and misc changes **bumps only the patch**

### version | 5.2.1 ###

* #### Updated | 2020-04-03 ####

* #### Created | 2014-10-21 ####

## Credits

`Maricopa Association of Governments (MAG) and the MAG member agencies`

## Site URL's

* [MAG Mapping](https://maps.azmag.gov/)

## Documentation

A detailed list of the changes and fixes in each version update can be found in the `CHANGELOG.md`.

* [CHANGELOG](CHANGELOG.md)

## Technologies

* Bootstrap v3.3.7
* Modernizr v3.5.0
* Font-Awesome v4.7
* jQuery v3.3.1
* Google Analytics
* Grunt v1.0.4

## Process

## Resources

## Dependencies

* [git-labelmaker](https://github.com/himynameisdave/git-labelmaker?utm_source=hashnode.com)

#### ***Reference Example Sites***

## Legal Disclaimer

The Maricopa Association of Governments provides the data within these pages as a public resource of general information for use "as is." The TMaricopa Association of Governments GIS (Geographic Information System) departments provides this information with the understanding that it is not guaranteed to be accurate, correct or complete and any conclusions drawn from such information are the sole responsibility of the user. Further, the Maricopa Association of Governments GIS departments makes no warranty, representation or guaranty as to the content, sequence, accuracy, timeliness or completeness of any of the spatial or database information provided herein. While every effort has been made to ensure the content, sequence, accuracy, timeliness or completeness of materials presented within these pages, the Maricopa Association of Governments GIS Departments assumes no responsibility for errors or omissions, and explicitly disclaims any representations and warranties, including, without limitation, the implied warranties of merchantability and fitness for a particular purpose. The Maricopa Association of Governments GIS Departments shall assume no liability for:

    1. Any errors, omissions, or inaccuracies in the information provided, regardless of how caused; or
    2. Any decision made or action taken or not taken by viewer in reliance upon any information or data furnished hereunder.

Availability of the Maricopa Association of Governments Map Server is not guaranteed. Applications, servers, and network connections may be unavailable at any time for maintenance or unscheduled outages. Outages may be of long duration. Users are cautioned to create dependencies on these services for critical needs.

THE FOREGOING WARRANTY IS EXCLUSIVE AND IN LIEU OF ALL OTHER WARRANTIES OF MERCHANTABILITY, FITNESS FOR PARTICULAR PURPOSE AND/OR ANY OTHER TYPE WHETHER EXPRESSED OR IMPLIED. In no event shall Maricopa Association of Governments become liable to users of these data, or any other party, for any loss or direct, indirect, special, incidental or consequential damages, including, but not limited to, time, money or goodwill, arising from the use or modification of the data.

To assist the Maricopa Association of Governments in the maintenance and/or correction of the data, users should provide the Maricopa Association of Governments GIS Departments with information concerning errors or discrepancies found in using the data. Please use the e-mail contact address at the bottom of the affected web page.

Please acknowledge the Maricopa Association of Governments (MAG) GIS as the source when Map Server data is used in the preparation of reports, papers, publications, maps, or other products.

## Copyright and Licensing

Code released under the MIT license.

A copy of the license is available in the repository's `LICENSE` file.

* [LICENSE](LICENSE)


CHANGELOG.md
# CHANGELOG #
#### MAG Maps Home Page ####

[![Semver](http://img.shields.io/SemVer/2.0.0.png)](http://semver.org/spec/v2.0.0.html)

All notable changes to this project will be documented in this file.

Releases will be numbered with the following format:

**`<major>.<minor>.<patch>`**

And constructed with the following guidelines:

1. **MAJOR** version when you make incompatible API changes **bumps the major** resets minor and patch
2. **MINOR** version when you add functionality in a backwards-compatible manner **bumps the minor** resets patch
3. **PATCH** version when you make backwards-compatible bug fixes and misc changes **bumps only the patch**

**************************************************

#### Types of Changes ####

* `Added` for new features
* `Changed` for changes in existing functionality
* `Deprecated` for soon-to-be removed features
* `Removed` for now removed features
* `Fixed` for any bug fixes
* `Security` in case of vulnerabilities

[Source](http://keepachangelog.com/en/1.0.0/)

***************************************************

## Version 5.2.0 | 2020-04-02


## Version 5.1.3 | 2020-01-29

* Changed intro to training page.
* Added Cache reset using querystring parameters to training and history page

## Version 5.1.2 | 2020-01-16

* Changed style for embedModal
* Added MapLIT to embedModal
* Added Cache reset using querystring parameters

## Version 5.1.1 | 2020-01-16

* Removed one training date that was wrong

## Version 5.1.0 | 2020-01-16

* Update training dates for 2020
* added new share links feature

## Version 5.0.18 | 2019-11-01

* Removed Heat Relief from map page
* Renamed Arizona Employment to Arizona Employer
* Added redirect URL to AZ Employer

## Version 5.0.17 | 2019-10-09

* Changed Employment Logo
* Added SASS

## Version 5.0.16 | 2019-07-08
## Version 5.0.15 | 2019-06-05
## Version 5.0.14 | 2019-04-25
## Version 5.0.13 | 2019-04-13

## Version 5.0.12 | 2019-03-26

* Removed training on west valley 3/28 per Merry

## Version 5.0.11 | 2019-03-04

* Removed training on 3/6 per Merry

## Version 5.0.10 | 2019-02-08

* Fixed github issue #20

## Version 5.0.9 | 2019-01-23

* Added new training dates

## Version 5.0.8 | 2019-01-10

* Added to the build process
* Removed old training dates
* Added new training dates

## Version 5.0.7 | 2018-11-30

* Added Schools Data Center to maps
* Changed Neighborhood Viewer icon
* Added Urban Area Information Viewer

## Version 5.0.6 | 2018-04-24

* Changed text on training page
* Fixed Links for Land Use and Projections - Not Https yet
* Removed Regional Employment Viewer Info
* Changed 'State Demographic Viewer' to 'Arizona Demographic Viewer'
* Changged 'State Employment Viewer' to 'Arizona Employment Viewer'
* Updated release history for Arizona Employment viewer

## Version 5.0.5 | 2018-04-20

* updated Release History info

## Version 5.0.4 | 2018-04-20

* Edited viewer box sizes and changed banner overlays

## Version 5.0.3 | 2018-04-19

* Fixed carousel-image css
* Fixed Facebook sharing button info
* Changed Training schedule page

## Version 5.0.2 | 2018-04-19

* fix readon links - not https

## Version 5.0.1 | 2018-04-19

* fixed facebook link
* Changed http to https for security

## Version 5.0.0 | 2018-04-19

* page redesign!
* added Google analytics tracking to buttons

## Version 4.7.7 | 2018-04-16

* Removed Spine Study Map from page
* add git-labels.json and to Github

## Version 4.7.6 | 2018-02-26

* removed regional demographics link
* updated details page

## Version 4.7.5 | 2018-02-14

* update training schedule links
* re-enabled training buttons for East and West Valley locations

## Version 4.7.4 | 2018-02-02

* temp disable of training buttons for East and West Valley locations

## Version 4.7.3 | 2018-01-26

* added video link to page
* update training links
* add ATP (Active Transportation) links
* change RAD link

## Version 4.7.2 | 10/19/2017

* updates to Regional Activity & Destination Viewer Basemap
* addition of State Employment Viewer

## Version 4.7.1 | 08/30/2017

* updates to Regional Activity & Destination Viewer

## Version 4.7.0 | 08/24/2017

* update icons for viewers
* moved `View Map` buttons
* Updated titles
* Moved `Subscribe for Update` button
* Moved `Register for Training` button

## Version 4.6.6 | 08/22/2017

* fix navigation buttons
* changed http://ims.azmag.gov/ to http://maps.azmag.gov/
* fixed some grunt tasks

## Version 4.6.5 | 08/22/2017

* Update to Employment Viewer - bug fix

## Version 4.6.4 | 08/07/2017

* Update to Employment Viewer 2016 Data

## Version 4.6.3 | 07/24/2017

* Add Regional Activity & Destination Viewer (RAD)
* Make icons into links also
* Update details page
* Add Grunt task to update index.html page on build

## Version 4.6.2 | 07/19/2017

* fix Grunt Build errors

## Version 4.6.1 | 07/19/2017

* update files
* Remove Victim Services Map Viewer
* update Grunt files

## Version 4.6.0 | 05/24/2017

* update to files only!
* NO code changes!
* removed node_modules from git
* created src folder
* created build folder

## Version 4.5.3 | 04/14/2017

* updates for State Demographic Viewer
* updates for Regional Demographic Viewer
* updates for Land Use Viewer
* updates for Employment Viewer

## Version 4.5.2 | 03/21/2017

* automated the copy write update info
* updated copy write info
* updates for State Demographics Viewer

## Version 4.5.0 | 02/09/2017

* added Spine Study Map Viewer to page

## Version 4.4.17 | 01/12/2017

* update to MAG Bikeways Viewer v3.2.0

## Version 4.4.16 | 01/09/2017

* update the training buttons on the map launch page

## Version 4.4.15 | 09/27/2016

* updates to employment viewer

## Version 4.4.14 | 06/28/2016

* updates to Projections Viewer

## Version 4.4.13 | 06/24/2016

* updated MAG Region Demographic Map Viewer
* updated State Demographic Map Viewer

## Version 4.4.12 | 06/16/2016

* updated MAG Employment Map Viewer

## Version 4.4.11 | 06/15/2016

* updated MAG Region Demographic Map Viewer

src/js/main.js
/* ========================================================================
 * Maricopa Association of Governments
 * JS document
 * @project     MAG Main Map Page
 * ========================================================================
 * @file        main.js
 * @summary     main JavaScript files for project
 * ========================================================================
 */

$(document).ready(function () {
    "use strict";

    // Links for social sharing
    var mapsLink = '<a href="https://maps.azmag.gov/" target="_blank" title="MAG Maps"><img src="https://maps.azmag.gov/images/share/MAG-maps-share-icon.png" alt="MAG Maps Icon" height="120"></a>';
    var demoLink = '<a href="https://geo.azmag.gov/maps/azdemographics/" target="_blank" title="Arizona Demographics"><img src="https://maps.azmag.gov/images/share/demographics-share-icon.png" alt="MAG Demographics Icon" height="120"></a>';
    var employLink = '<a href="https://geo.azmag.gov/maps/azemployer/" target="_blank" title="Arizona Employer"><img src="https://maps.azmag.gov/images/share/Employment-share-icon.png" alt="MAG Employer Icon" height="120"></a>';
    var bikeLink = '<a href="https://geo.azmag.gov/maps/bikemap/" target="_blank" title="MAG Bikeways"><img src="https://maps.azmag.gov/images/share/bikeways-share-icon.png" alt="MAG Bikemap Icon" height="120"></a>';
    var radLink = '<a href="https://rad.azmag.gov/" target="_blank" title="MAG RAD"><img src="https://maps.azmag.gov/images/share/RAD-share-icon.png" alt="MAG RAD Icon" height="120"></a>';
    var mapLITLink = '<a href="http://readonarizona.org/data-tools/maplit/" target="_blank" title="MAG MapLIT"><img src="https://maps.azmag.gov/images/share/MapLIT_smaller.png" alt="MAG MapLIT icon" height="120"></a>';


    //*** Load page sections

    //Load nav bar then run function that adds active to nav link
    $("#navbar-section").load("views/section-navbar.html", function () {
        $("#magNavbar .nav li a").each(function () {
            if ($(this).prop("href") === window.location.href) {
                $(this).parent("li").addClass("active");
            }
        });

        /*twitter share button href*/
        var baseUrl = 'https://twitter.com/intent/tweet';
        var text = "MAG%20%7C%20Mapping%20Center";
        var thisPageUrl = "https://maps.azmag.gov/";
        var hashTag = "MAGmaps";
        var via = "MAGregion";

        var twitterHref = (baseUrl + "?text=" + text + "&url=" + thisPageUrl + "&hashtag=" + hashTag + "&via=" + via);

        $('#TWshareButton').attr('href', twitterHref);
    });

    $("#carousel-section").load("views/section-carousel.html");
    $("#maps-section").load("views/section-maps.html");
    $("#video-section").load("views/section-video.html");
    $(".footer-section").load("views/section-footer.html", function () {
        //*** terms binding
        $("#termsModal").load("views/modal-terms.html");
        //*** privacy binding
        $("#privacyModal").load("views/modal-privacy.html");
        //*** version binding
        $(".version").html("v5.2.1 | 2020-04-03");
        //*** copy write binding
        $(".copyright").html("2020");

        //*** embed binding
        $("#embedModal").load("views/modal-embed.html", function () {
            $("#copy1")[0].setAttribute('value', mapsLink);
            $("#copy2")[0].setAttribute('value', employLink);
            $("#copy3")[0].setAttribute('value', demoLink);
            $("#copy4")[0].setAttribute('value', bikeLink);
            $("#copy5")[0].setAttribute('value', radLink);
            $("#copy6")[0].setAttribute('value', mapLITLink);
        });

    });
    $("#pastUpdates").load("views/section-pastUpdates.html");
    $("#mostRecent").load("views/section-recentUpdates.html");

    //*** Modals ***
    $("#stateDemographics").load("views/modal-stateDemographics.html");
    $("#rad").load("views/modal-rad.html");
    $("#readOnAZ").load("views/modal-readOnAZ.html");
    $("#employment").load("views/modal-employment.html");
    $("#stateEmployment").load("views/modal-stateEmployment.html");
    $("#bikeways").load("views/modal-bikeways.html");
    $("#landUse").load("views/modal-landUse.html");
    $("#projections").load("views/modal-projections.html");
    $("#neighborhood").load("views/modal-neighborhood.html");
    // $("#spineStudy").load("views/modal-spineStudy.html");
    $("#atp").load("views/modal-atp.html");
    $("#tip").load("views/modal-tip.html");
    $("#schoolsDataCenter").load("views/modal-schoolsDataCenter.html");
    $("#urbanAreas").load("views/modal-urbanAreas.html");
    $("#heatRelief").load("views/modal-heatRelief.html");
    $("#healthcare").load("views/modal-healthcare.html");






});


/*
 ***********************************
 * social pluggins
 ***********************************
 */

// twitter widget js loading
! function (d, s, id) {
    var js, fjs = d.getElementsByTagName(s)[0],
        p = /^http:/.test(d.location) ? "http" : "https";
    if (!d.getElementById(id)) {
        js = d.createElement(s);
        js.id = id;
        js.src = p + "://platform.twitter.com/widgets.js";
        fjs.parentNode.insertBefore(js, fjs);
    }
}(document, "script", "twitter-wjs");

//LinkedIn
$(document).on('click', '#INshareButton', function () {
    window.open("https://www.linkedin.com/shareArticle?url=http://maps.azmag.gov/", "shareLinkedIn", 'width=650, height=700');
});

// google +1
(function () {
    var po = document.createElement("script");
    po.type = "text/javascript";
    po.async = true;
    po.src = "https://apis.google.com/js/platform.js";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(po, s);
})();

$(document).on('click', '#GPlusShareButton', function () {
    window.open("//plus.google.com/share?url=http%3A%2F%2Fmaps.azmag.gov%2F", 'shareGooglePlus', 'width=400, height=700');
});

// facebook
//share dialog - use on('click') for it to work afte loading html
$(document).on('click', '#FBshareButton', function () {
    FB.ui({
        method: 'share',
        mobile_iframe: true,
        href: 'https://maps.azmag.gov/',
    }, function (response) {});
});

window.fbAsyncInit = function () {
    FB.init({
        appId: "929950963769905",
        cookie: true,
        xfbml: true,
        version: "v2.12"
    });
};

(function (d, s, id) {
    var js, fjs = d.getElementsByTagName(s)[0];
    if (d.getElementById(id)) {
        return;
    }
    js = d.createElement(s);
    js.id = id;
    js.src = "https://connect.facebook.net/en_US/sdk.js";
    fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));


$(document).on('click', '.shareButton button', function (e) {

    e.preventDefault();
    $(this).toggleClass('active');
    var shareBox = $(this).attr('data-target');

    if ($(this).hasClass('active')) {
        $(shareBox).show();
    }
});

$('body').mouseup(function (e) {
    var parentTarget = $(e.target).parents('#shareBox');

    if (e.target.id !== 'shareBox' && parentTarget.length < 1 && $(e.target).hasClass('shareButton') == false) {
        $('#shareBox').hide();
        $('.shareButton button').removeClass('active');
    }
});

//Open email window
function openNewWindow(obj) {
    "use strict";
    var thisHref = obj.href;
    window.open(thisHref, 'popup', 'width=600,height=600');
    // <!-- comments:uncomment // -->
    // ga("send", "event", "Click", "Email-Button", "Email-Button");
    // <!-- endcomments -->
}

//collapse one at a time
$(".toggle-links").click(function () {
    "use strict";
    $(".collapseGroup .collapse").collapse("hide");
    var showTarget = $(this).attr("href");

    $(showTarget).collapse("show");

    if (!$(this).hasClass("active")) {
        $(".toggle-links").removeClass("active");
        $(this).addClass("active");
    }
});

$(window).scroll(function () {
    "use strict";
    if ($(document).scrollTop() > 20) {
        $(".scrollTop").css("opacity", "1");
    } else {
        $(".scrollTop").css("opacity", "0");
    }

});

$(".scrollTop").click(function () {
    "use strict";
    window.scroll({
        top: 0,
        behavior: "smooth"
    });
});

// ------------------------------
// GOOGLE ANALYTICS CTA TRACKING
// -----------------------------
$(document).on('click', '.track', function (e) {
    var text = $(this).text(); // Grab the text from the button
    // console.log(text);
    // <!-- comments:uncomment // -->
    // ga('send', 'event', 'Button', 'Click', text);
    // <!-- endcomments -->
});
